Draft: 設定画面の実装をしました

meithon opened the pull request:
- **feat(settings): implement settings screen with sub-routes**
- **refactor(routing): implement nested settings routes**

## 対応したこと

<!-- 対応したことを箇条書きでわかりやすく -->

## 対応 Issue
- https://github.com/KDG-Develop-Hub/social-mobile/issues/47
- https://github.com/KDG-Develop-Hub/social-mobile/issues/48
- https://github.com/KDG-Develop-Hub/social-mobile/issues/49
- https://github.com/KDG-Develop-Hub/social-mobile/issues/50
<!-- 該当するIssueを貼り付ける -->

## 関連資料

<!-- 対応する中で参考にしたWebサイトがあれば、何の対応に対して参考にしたか明記しながら、URLを貼る -->

## タスク
- Screenの定義
  - [x] プロフィール
  - [x] アカウント
  - [x] 見た目
  - [x] 言語
- Routing
  - [x] プロフィール
  - [x] アカウント
  - [x] 見た目
  - [x]言語
- Figmaのデザインに合わせる
  - [x] プロフィール
  - [x] アカウント
  - [x] 見た目
  - [x] 言語
- [ ] i18n
- mockやMobileなどのAPIとの結合
  - [ ] プロフィール
  - [ ] アカウント
  - [ ] 見た目
  - [ ] 言語
<!-- 対応しきれなかった部分、自分では実装できない部分をchecklistで箇条書き -->

## 画面キャプチャ

<!-- UIの新規実装の場合、スクショを貼る。UIの修正の場合は修正後スクショと、あればbeforeのスクショもあるとレビューしやすい -->

meithon committed:
feat(settings): implement settings screen with sub-routes

- Add settings screen with search functionality
- Create sub-screens for profile, account, appearance and language settings
- Add route configurations for settings navigation
- Implement form controls for profile editing
- Add theme and font size controls
- Temporarily disable dark theme for development

meithon committed:
refactor(routing): implement nested settings routes

- Remove profile edit tile from settings screen
- Add nested routes for settings screens (profile, account, appearance, language)
- Implement route data classes for settings sub-routes
- Update route constants organization

meithon commented on lib/utils/routes/app_router.dart:
```diff
@@ -66,6 +66,20 @@ GoRouter goRouter(GoRouterRef ref) {
       routes: [
         TypedGoRoute<SettingsScreenRouteData>(
           path: AppRoutes.settings,
+          routes: [
+            TypedGoRoute<SettingsProfileRouteData>(
+              path: AppRoutes.settingsProfilePath,
+            ),
+            TypedGoRoute<SettingsAccountRouteData>(
+              path: AppRoutes.settingsAccountPath,
+            ),
+            TypedGoRoute<SettingsAppearanceRouteData>(
+              path: AppRoutes.settingsAppearancePath,
+            ),
+            TypedGoRoute<SettingsLanguageRouteData>(
+              path: AppRoutes.settingsLanguagePath,
+            ),
+          ],
```
[question] ここで Nested Routing にするのって微妙？
lib/utils/routes/routes.dart にある `AppRoutes`classとの兼ね合い的なものや、@TypedGoRouteでNested Routingを定義できるようなので気になりました

Haru-Kobayashi073 replied:
@TypedGoRoute配下でネストさせることと違いがなさそうであれば、1箇所にまとめたいから今定義してくれてる部分で大丈夫！

meithon committed:
refactor(settings): reorganize profile and account settings screens

- Add prefixIcon support to TextFieldWidget
- Implement form validation for profile settings
- Create separate account deactivation component
- Update padding in TextFieldWidget to be symmetric

meithon committed:
refactor(settings): split settings screens into separate files

meithon committed:
feat(ui): Improved look of the Profile screen.

- Make maxLength and maxLines optional in TextFieldWidget
- Add onChanged callback to TextFieldWidget
- Update padding and styling in TextFieldWidget
- Implement new profile settings screen with SNS field management
- Add profile-related routes to app router

meithon committed:
ui(account): Improve look of account screen

meithon committed:
ui(appearance): Implement UI from Figma design

meithon committed:
ui(language): Implement state of language screen
